phylokit/distance.py

```diff
@@ -0,0 +1,142 @@
+# Tree distance metrics.
+
+import numba
+import numpy as np
+
+
+@numba.njit(cache=True)
+def _branch_length(parent, time, u):
+    ret = 0
+    p = parent[u]
+    if p != -1:
+        ret = time[p] - time[u]
+    return ret
+
+
+def branch_length(tree, u):
+    """
+    Returns the length of the branch (in units of time) joining the specified
+    node to its parent.
+
+    :param int u: The node ID.
+    :return : The length of the branch.
+    :rtype : float
+    """
+    num_nodes = tree.tree_sequence.num_nodes
+    if u < 0 or u > num_nodes:
+        raise ValueError("Node index out of bounds")
+    return _branch_length(
+        tree.parent_array, tree.tree_sequence.tables.nodes.time, u
```

Comment:
Ah, so we don't have a tree.time_array yet, and accessing the time array like this will murder performance. There's an issue on tskit to track adding the arrays, https://github.com/tskit-dev/tskit/issues/1322